runtime/integration-tests/src/generic/cases/lp/pool_management.rs

```diff
@@ -0,0 +1,249 @@
+
+	#[allow(non_camel_case_types)]
+	pub struct TestCurrency;
+	impl CurrencyInfo for TestCurrency {
+		const CUSTOM: CustomMetadata = CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::LiquidityPools,
+			permissioned: false,
+			mintable: false,
+		};
+		const DECIMALS: u32 = 12;
+		const ED: Balance = 10_000_000_000;
+		const ID: CurrencyId = CurrencyId::ForeignAsset(200_001);
+		const SYMBOL: &'static str = "FRAX";
+	}
+
+	env.deploy(
+		"ERC20",
+		"test_erc20",
+		Keyring::Admin,
+		Some(&[Token::Uint(Uint::from(TestCurrency::DECIMALS))]),
+	);
+
+	let test_erc20_address = env.deployed("test_erc20").address();
+
+	env.parachain_state_mut(|| {
+		register_currency::<T, USDC>(Some(utils::lp_asset_location::<T>(test_erc20_address)));
+
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_currency(
+			OriginFor::<T>::signed(Keyring::Alice.into()),
+			TestCurrency::ID
+		));
+
+		utils::process_outbound::<T>()
+	});
+
+	let index = GeneralCurrencyIndexOf::<T>::try_from(TestCurrency::ID).unwrap();
+
+	// Verify the  test currencies are correctly added to the pool manager
+	assert_eq!(
+		Decoder::<H160>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"currencyIdToAddress",
+				Some(&[Token::Uint(Uint::from(index.index))])
+			)
+			.unwrap()
+			.value
+		),
+		test_erc20_address
+	);
+
+	assert_eq!(
+		Decoder::<Balance>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"currencyAddressToId",
+				Some(&[Token::Address(test_erc20_address)]),
+			)
+			.unwrap()
+			.value
+		),
+		index.index
+	);
+}
+
+fn add_pool<T: Runtime>() {
+	let mut env = super::setup::<T>(|_| {});
+	const POOL: PoolId = 1;
+
+	env.parachain_state_mut(|| {
+		crate::generic::utils::pool::create_one_tranched::<T>(
+			Keyring::Admin.into(),
+			POOL,
+			LocalUSDC::ID,
+		);
+
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_pool(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL,
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>()
+	});
+
+	let creation_time = env.parachain_state(<pallet_timestamp::Pallet<T> as TimeAsSecs>::now);
+
+	// Compare the pool.created_at field that is returned
+	assert_eq!(
+		Decoder::<Uint>::decode(
+			&env.view(
+				Keyring::Alice,
+				"pool_manager",
+				"pools",
+				Some(&[Token::Uint(Uint::from(POOL))]),
+			)
+			.unwrap()
+			.value
+		),
+		Uint::from(creation_time)
+	);
+
+	env.pass(Blocks::ByNumber(1));
+
+	env.parachain_state_mut(|| {
+		assert_ok!(pallet_liquidity_pools::Pallet::<T>::add_pool(
+			OriginFor::<T>::signed(Keyring::Admin.into()),
+			POOL,
+			Domain::EVM(EVM_DOMAIN_CHAIN_ID)
+		));
+
+		utils::process_outbound::<T>()
+	});
+
+	// TODO: Actually find the revert event here.
+	//
+	// NOTE: That is really relevant for the router too.
+	//     We need to check `Pending` and see for errors
+	//     unfortuntately and then error in the router instead of returning success.
+
```

Comment:
Could you elaborate on the comment? I cannot follow because I don't know which event you are referring to.

Reply:
The problem is that the EVM is never returning an error. It is always Ok, even if reverted. We need to check the core return type of the evm. It is stored in the Pending storage of the evm on the frontier side.